daml-script/export/src/main/scala/com/daml/script/export/TreeUtils.scala

```diff
@@ -237,16 +237,31 @@ object TreeUtils {
   object Command {
     def fromTree(tree: TransactionTree): Seq[Command] = {
       val contractKeys = mutable.HashMap.empty[ContractId, Value]
+      def addContractKey(createdEvent: CreatedEvent): Unit = {
```

Comment:
I think I forgot how we handle ExerciseByKey. Can you confirm that my understanding upon rereading the code is correct:
We turn each exercise into an exerciseByKey provided that the contract has been created (and has a key) in the current transaction and has not been archived since.

Reply:
Correct, with one exception: If the exercise directly follows the create that created the contract, then we issue a createAndExercise instead of create followed by exerciseByKey.